util/git/client.go

```diff
@@ -185,7 +171,11 @@ func (m *nativeGitClient) LsRemote(revision string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	refs, err := remote.List(&git.ListOptions{Auth: m.auth})
+	auth, err := newAuth(m.repoURL, m.creds)
```

Comment:
I'd love to get rid of auth and only use Git CLI

Reply:
When we refactor "application discovery" of a git repository, I think we will no longer have the need for `LsFiles` since I think we would instead do a checkout of the revision, and a direct filesystem traverse instead of `ls-files` . I think it would be much quicker too, and we would have some caching opportunities that we don't currently have.